### PR TITLE
Improve/fix drag-and-drop

### LINKED ISSUE #154: I want to disable horizontal and vertical scrolling when dragging an event somehow
Whenever trying to drag the event, calendar view swipe behavior and vertical scrolling behavior work at the same time. It looks like so uncomfortable to drag the event.
Is there any way to disable horizontal and vertical scrolling when dragging the event?
Thanks.

### PR DESCRIPTION
**Pull request by pdpino**
Fixes #154 

* Make drag and drop much smoother (see demo in the issue)
* Added [react-native-gesture-handler](https://docs.swmansion.com/react-native-gesture-handler/docs/) and [react-native-reanimated](https://docs.swmansion.com/react-native-reanimated/docs/) libraries
  * These libs work much better than react-native's [gesture responder system ](https://reactnative.dev/docs/gesture-responder-system) and [Animated](https://reactnative.dev/docs/animated) APIs
  * This will allow much better animations and interactions, for example #58 or #130

Lmk what you think!

### CODE REVIEW

**Comment by hoangnm** [comment]
Hi @pdpino, I tried to run on local but the calendar is frozen and I couldn't perform any action, anything specific I need to set up when running on local?

File: src/utils-gestures.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    const composedGesture = Gesture.Exclusive(longPressGesture, pressGesture);
+    return (
+      <GestureDetector gesture={composedGesture}>
+        {/* eslint-disable-next-line react/jsx-props-no-spreading */}
+        <Component {...other} />
+      </GestureDetector>
+    );
+  };
+  return viewWithTouchable;
```
Comment (hoangnm) [inline src/utils-gestures.js:51]: wondering how do you know this patch?

**Comment by hoangnm** [comment]
nvm, I can make it work now.

**Comment by pdpino** [comment]
@hoangnm only installing peer dependencies react-native-gesture-handler v2 and react-native-reanimated v2, following their instructions

File: src/__tests__/Event.test.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    const position = { top: 10, left: 0, width: 40, height: 50 };
+
+    render(<Event event={mockEvent} onDrag={onDragMock} position={position} />);
+    fireGestureHandler(getByGestureTestId('dragGesture'), buildDragGesture());
+
+    expect(onDragMock).toHaveBeenCalledTimes(1);
+    expect(onDragMock).toHaveBeenCalledWith(
+      mockEvent,
+      position.left + position.width / 2 + 7,
+      position.top + 52,
```
Comment (hoangnm) [inline src/__tests__/Event.test.js:37]: can we have a clear naming for these values?

**Comment by hoangnm** [comment]
Drag & drop are really smoother, this would be a great improvement.

File: src/utils-gestures.js (hunk 1 above)
Comment (pdpino) [reply src/utils-gestures.js:51]: I just did it myself :sweat_smile:

File: src/__tests__/Event.test.js (hunk 2 above)
Comment (pdpino) [reply src/__tests__/Event.test.js]: @hoangnm just updated

File: src/utils-gestures.js (hunk 1 above)
Comment (hoangnm) [reply src/utils-gestures.js:79]: great!

**Comment by hoangnm** [comment]
I'm thinking to use this feature as a version 2 and I'll bump the current code base to v1.

**Comment by pdpino** [comment]
Some users might be using in production: yes

But I think the API might still change, e.g. some props:

* `numberOfDays` for the horizontal-pinch gesture
* `startHour, hoursInDisplay, timeStep` for the vertical-pinch gesture
* Or others: https://github.com/hoangnm/react-native-week-view/issues/58, https://github.com/hoangnm/react-native-week-view/issues/128, https://github.com/hoangnm/react-native-week-view/issues/141#issuecomment-942321193, https://github.com/hoangnm/react-native-week-view/issues/186

I prefer keeping 0.x, we can worry less about backward compatibility, and finish some of these new features first.

**Comment by hoangnm** [comment]
ok @pdpino , that makes sense.

**Comment by hoangnm** [comment]
Great PR, merging.

**Pull request merged**